modules/apps/headless/headless-web-experience/headless-web-experience-test/src/testIntegration/java/com/liferay/headless/web/experience/resource/v1_0/test/StructuredContentResourceTest.java

```diff
@@ -15,15 +15,233 @@
+
+		assertResponseCode(
+			404,
+			invokeGetStructuredContentResponse(postStructuredContent.getId()));
+
+	}
+
+	@Override
+	@Test
+	public void testGetStructuredContent() throws Exception {
+		StructuredContent postStructuredContent =
+			invokePostContentSpaceStructuredContent(
+				testGroup.getGroupId(), randomStructuredContent());
+
+		StructuredContent getStructuredContent = invokeGetStructuredContent(
+			postStructuredContent.getId());
+
+		assertEquals(postStructuredContent, getStructuredContent);
+		assertValid(getStructuredContent);
+	}
+
+	@Override
+	@Test
+	public void testPostContentSpaceStructuredContent() throws Exception {
+		StructuredContent randomStructuredContent = randomStructuredContent();
+
+		StructuredContent postStructuredContent =
+			invokePostContentSpaceStructuredContent(
+				testGroup.getGroupId(), randomStructuredContent);
+
+		assertEquals(randomStructuredContent, postStructuredContent);
+		assertValid(postStructuredContent);
+	}
+
+	@Override
+	@Test
+	public void testPutStructuredContent() throws Exception {
+		StructuredContent postStructuredContent =
+			invokePostContentSpaceStructuredContent(
+				testGroup.getGroupId(), randomStructuredContent());
+
+		StructuredContent randomStructuredContent = randomStructuredContent();
+
+		StructuredContent putStructuredContent = invokePutStructuredContent(
+			postStructuredContent.getId(), randomStructuredContent);
+
+		assertEquals(randomStructuredContent, putStructuredContent);
+		assertValid(putStructuredContent);
+
+		StructuredContent getStructuredContent = invokeGetStructuredContent(
+			putStructuredContent.getId());
+
+		assertEquals(randomStructuredContent, getStructuredContent);
+		assertValid(getStructuredContent);
+	}
+
+	@Override
+	protected void assertEquals(
+		StructuredContent structuredContent1,
+		StructuredContent structuredContent2) {
+
+		Assert.assertEquals(
+			structuredContent1.getContentSpace(),
+			structuredContent2.getContentSpace());
+
+		Assert.assertEquals(
+			structuredContent1.getContentStructureId(),
+			structuredContent2.getContentStructureId());
+
+		Assert.assertEquals(
+			structuredContent1.getDescription(),
+			structuredContent2.getDescription());
+
+		Assert.assertEquals(
+			structuredContent1.getTitle(), structuredContent2.getTitle());
+	}
+
+	protected void assertValid(StructuredContent structuredContent) {
+		boolean valid = false;
+
+		if ((structuredContent.getDateCreated() != null) &&
+			(structuredContent.getDateModified() != null) &&
+			(structuredContent.getId() != null) &&
+			Objects.equals(
+				structuredContent.getContentSpace(), testGroup.getGroupId())) {
+
+			valid = true;
+		}
+
+		Assert.assertTrue(valid);
+	}
+
+	@Override
+	protected StructuredContent randomStructuredContent() {
+		return new StructuredContent() {
+			{
+				contentSpace = testGroup.getGroupId();
+				contentStructureId = _ddmStructure.getStructureId();
+				dateCreated = RandomTestUtil.nextDate();
```

Comment:
dateCreated, dateModified, datePublished, id and lastReviewed should be null, right?